app/scripts/metamask-controller.js

```diff
@@ -4395,7 +4441,7 @@ export default class MetamaskController extends EventEmitter {
     const accounts = this.accountsController.listAccounts();
 
     const { identities } = this.preferencesController.store.getState();
-    return { unlockedAccount, identities, accounts };
+    return { unlockedAccount: account.address, identities, accounts };
```

Comment:
Nit: was this change needed? I'd expect this to be equivalent:
```suggestion
    return { unlockedAccount, identities, accounts };
```

Reply:
If I remember correctly, the address returned directly by the keyring can be lowercase, while the one coming from accounts controller is checksummed (or the other way around). Since we don't know how the specific keyring will return the address, I thought that it would be safer this way and would fix unit tests

Reply:
Previously we were using `this.keyringController.addNewAccountForKeyring`, which returns lower-cased addresses. We lower-case the address above to normalize it.

Reply:
Yes exactly, since that is now inside `withKeyringForDevice`' callback we can't call `KeyringController`. We can normalize it

Though, it looks like it is only used for tests

Reply:
Normalized in [01ca220](https://github.com/MetaMask/metamask-extension/pull/25435/commits/01ca22023cea120e40d81dbf5d00f34217be569d)